File: Demo/Sources/ViewControllers/Transition/Root/TransitionRootViewController.m
//
//  Globus
//

#import "TransitionRootViewController.h"
#import "TransitionChildViewController.h"

@implementation TransitionRootViewController

#pragma mark - Actions

- (IBAction)pressedPresent:(id)sender {
    TransitionChildViewController* vc = [TransitionChildViewController instantiate];
    GLBNavigationViewController* nvc = [GLBNavigationViewController viewControllerWithRootViewController:vc];
    nvc.transitionNavigation = self.transition;
    nvc.transitionModal = self.transition;
    [self presentViewController:nvc animated:YES completion:nil];
}

@end

File: Demo/Sources/ViewControllers/Transition/Root/TransitionRootViewController.swift
//
//  Globus
//

import Globus

class TransitionRootViewController: TransitionViewController {
    
    // MARK - Action
    
    @IBAction internal func pressedPresent(_ sender: Any) {
        let vc = TransitionChildViewController.instantiate()!
        let nvc = GLBNavigationViewController(rootViewController: vc)
        nvc.transitionNavigation = self.transition
        nvc.transitionModal = self.transition
        self.present(nvc, animated: true, completion: nil)
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "TransitionRootViewController-Swift"
    }
    
}
